mirage-nat/unikernel.ml

```diff
@@ -46,16 +53,25 @@ module Main (R : Mirage_random.S) (M : Mirage_clock.MCLOCK) (P : Mirage_clock.PC
                         (match e with `Local -> "local" | `Gateway -> "gateway"));
             Lwt.return_unit
           | Ok dst ->
+            let more = ref [] in
             E.write eth dst `IPv4 (fun b ->
                 match Nat_packet.into_cstruct packet b with
-                | Ok n -> n
+                | Ok (n, adds) -> more := adds ; n
```

Comment:
is `n` guaranteed to be <= `sizeof b` ? should we `assert` ?

Reply:
it is (due to me knowing the mirage_nat implementation) ;)